app/parser/src/parse.server.ts

```diff
@@ -239,26 +245,30 @@ export async function parse(rawArgs: string[]) {
     "Error hydrating commit tree"
   )
 
-  let outPath = args.out ?? `${repoName}_${branchName}.json`
+  const defaultOutPath = resolve(__dirname, "..", ".temp", repoName, `${branchName}.json`)
+  let outPath = resolve(args.out ?? defaultOutPath)
   if (!isAbsolute(outPath))
-    outPath = resolve(truckroot_or_cwd, outPath)
+    outPath = resolve(cwd, outPath)
 
   const authorUnions = await loadTruckConfig(repoDir)
+  const data = {
+    repo: repoName,
+    branch: branchName,
+    commit: hydratedRepoTree,
+    authorUnions: authorUnions,
+  }
   await describeAsyncJob(
     () =>
-      writeRepoToFile(outPath, {
-        repo: repoName,
-        branch: branchName,
-        commit: hydratedRepoTree,
-        authorUnions: authorUnions,
-      }),
+      writeRepoToFile(outPath, data),
```

Comment:
Should we both write data to a file and return it? Is the written file mostly for debugging purposes?

Reply:
For debugging and chaching later on